llvm/lib/Target/AMDGPU/AMDGPUCodeGenPassBuilder.cpp

```diff
@@ -106,3 +135,12 @@ Error AMDGPUCodeGenPassBuilder::addInstSelector(AddMachinePass &addPass) const {
   addPass(SILowerI1CopiesPass());
   return Error::success();
 }
+
+bool AMDGPUCodeGenPassBuilder::isPassEnabled(const cl::opt<bool> &Opt,
```

Comment:
is this pattern specific to AMDGPU? perhaps it should go in a more general place

also in general, is it necessary to have so many flags that force enable/disable passes? e.g. the middle-end optimizer doesn't have so many flags that force enable/disable passes

Reply:
I guess most everything gets introduced with a flag and then the flag never goes away

Reply:
I'd prefer if we avoided adding this method, since it makes it too easy code-wise to keep around these flags. This might require cleaning up some flags so that we don't have to inline this method in so many places. IMO adding code noise around these flags is a good thing since these flags are undesirable.

Reply:
> IMO adding code noise around these flags is a good thing since these flags are undesirable.

I'm not moved by this argument at all